processing/src/main/java/org/apache/druid/segment/nested/NestedFieldLiteralColumnIndexSupplier.java

```diff
@@ -336,17 +336,21 @@ public BitmapColumnIndex forValue(@Nullable String value)
         @Override
         public double estimateSelectivity(int totalRows)
         {
-          return (double) getBitmap(
-              localDictionary.indexOf(stringDictionary.indexOf(StringUtils.toUtf8ByteBuffer(value)))
-          ).size() / totalRows;
+          final int globalId = stringDictionary.indexOf(StringUtils.toUtf8ByteBuffer(value));
+          if (globalId < 0) {
+            return 0.0;
+          }
+          return (double) getBitmap(localDictionary.indexOf(globalId)).size() / totalRows;
         }
 
         @Override
         public <T> T computeBitmapResult(BitmapResultFactory<T> bitmapResultFactory)
         {
-          return bitmapResultFactory.wrapDimensionValue(
-              getBitmap(localDictionary.indexOf(stringDictionary.indexOf(StringUtils.toUtf8ByteBuffer(value))))
-          );
+          final int globalId = stringDictionary.indexOf(StringUtils.toUtf8ByteBuffer(value));
+          if (globalId < 0) {
+            return bitmapResultFactory.wrapDimensionValue(bitmapFactory.makeEmptyImmutableBitmap());
+          }
+          return bitmapResultFactory.wrapDimensionValue(getBitmap(localDictionary.indexOf(globalId)));
```

Comment:
We seem to do the same general checks between these two methods on a regular basis.  Maybe we should add an `AtomicReference<>` to the return object and another method that builds and stores an object.  It seems like the 3 possible outcomes are

1. We are looking for `null`
2. We are looking for a value that doesn't exist
3. We are looking for a value that does exist

If the Reference was to an object that just maintained those 3 states, we could do that once, cache it and then implement each of these methods off of those states.  I'm making this suggestion as I think this structuring might be less bug prone (and potentially even faster as I believe both methods get called in the course of a query, so not doing the same lookups multiple times could be a gain, though likely very small).

Reply:
estimate method is currently only called by search query with 'auto' indexes strategy, and only indirectly. I am starting to think this shouldn't actually be part of the `BitmapColumnIndex` interface and can instead be done entirely by the `Filter` instead...

Reply:
Or... maybe the search query logic could change to do it on its own?  Don't even do it in the filter?

Reply:
i might make sense to remain in `Filter` since the 'and'/'or' filters are the only ones that really do anything besides get the size of the bitmap and divide by number of rows.

I think refactoring this right now is a bit too much for this bug fix and I'd rather do as a follow-up, is that ok?

Reply:
ah saw the other comment, yeah will do this in follow-up